src/libraries/Common/src/System/Security/Cryptography/SlhDsa.cs

```diff
@@ -317,17 +306,84 @@ public bool TryExportPkcs8PrivateKey(Span<byte> destination, out int bytesWritte
         {
             ThrowIfDisposed();
 
-            // TODO: Once the minimum size of a PKCS#8 export is known, add an early return false.
+            // An SLH-DSA-SHA2-128s private key export with no attributes is 84 bytes. A buffer smaller than that cannot hold a
+            // PKCS#8 encoded key. If we happen to get a buffer smaller than that, it won't export.
+            const int MinimumPossiblePkcs8SlhDsaKey = 84;
 
-            throw new NotImplementedException("The PKCS#8 format is still under debate");
+            if (destination.Length < MinimumPossiblePkcs8SlhDsaKey)
+            {
+                bytesWritten = 0;
+                return false;
+            }
+
+            return TryExportPkcs8PrivateKeyCore(destination, out bytesWritten);
         }
 
         /// <summary>
-        ///  Exports the current key in a PEM-encoded representation of the PKCS#8 PrivateKeyInfo format.
+        ///   When overridden in a derived class, attempts to export the current key in the PKCS#8 PrivateKeyInfo format
+        ///   into the provided buffer.
         /// </summary>
+        /// <param name="destination">
+        ///   The buffer to receive the PKCS#8 PrivateKeyInfo value.
+        /// </param>
+        /// <param name="bytesWritten">
+        ///   When this method returns, contains the number of bytes written to the <paramref name="destination"/> buffer.
+        /// </param>
         /// <returns>
-        ///   A string containing the PEM-encoded representation of the PKCS#8 PrivateKeyInfo
-        ///   representation of the public-key portion of this key.
+        ///   <see langword="true" /> if <paramref name="destination"/> was large enough to hold the result;
+        ///   otherwise, <see langword="false" />.
+        /// </returns>
+        /// <exception cref="ObjectDisposedException">
+        ///   This instance has been disposed.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   An error occurred while exporting the key.
+        /// </exception>
+        protected virtual bool TryExportPkcs8PrivateKeyCore(Span<byte> destination, out int bytesWritten)
+        {
+            // Secret key size for SLH-DSA is at most 128 bytes so we can stack allocate it.
+            int secretKeySizeInBytes = Algorithm.SecretKeySizeInBytes;
+            Debug.Assert(secretKeySizeInBytes is <= 128 and >= 0);
+            Span<byte> buffer = stackalloc byte[secretKeySizeInBytes];
```

Comment:
Consider just stackalloc-ing 128 bytes and slicing it to the appropriate size. The JIT prefers const values in to stackalloc anyway.